### PR TITLE
external services api: check kind and fetch by ID

### PR DESCRIPTION
**Pull request by nicksnyder**
Backend changes for https://github.com/sourcegraph/sourcegraph/pull/1103
Part of https://github.com/sourcegraph/sourcegraph/issues/914

### CODE REVIEW

**Comment by nicksnyder** [comment]
merging to unblock

File: cmd/frontend/graphqlbackend/external_services.go [hunk 1]
```diff
@@ -11,6 +12,22 @@ import (
...
+var externalServiceKinds = map[string]struct{}{
+	"AWSCODECOMMIT":   struct{}{},
+	"BITBUCKETSERVER": struct{}{},
+	"GITHUB":          struct{}{},
+	"GITLAB":          struct{}{},
+	"GITOLITE":        struct{}{},
+	"PHABRICATOR":     struct{}{},
+}
+
+func checkKind(kind string) error {
```
Comment (slimsag) [inline cmd/frontend/graphqlbackend/external_services.go]: minor nit: `validateKind` seems like better naming to me for these types of functions

**Comment by slimsag** [comment]
+1

**Comment by nicksnyder** [comment]
Thanks for the review! Comments addressed in https://github.com/sourcegraph/sourcegraph/commit/61107a717e52a617c88200bc79c3485118440a96

**Pull request merged**